tests/mechanic/telemetry_test.py

```diff
@@ -148,30 +148,46 @@ def test_store_calculated_metrics(self, listdir_mock, open_mock, metrics_store_p
 class Client:
     def __init__(self, nodes=None, info=None, indices=None, transport_client=None):
         self.nodes = nodes
-        self._info = info
+        self._info = wrap(info)
         self.indices = indices
         if transport_client:
             self.transport = transport_client
 
     def info(self):
-        return self._info
+        return self._info()
 
 
 class SubClient:
     def __init__(self, stats=None, info=None, recovery=None):
-        self._stats = stats
-        self._info = info
-        self._recovery = recovery
+        self._stats = wrap(stats)
+        self._info = wrap(info)
+        self._recovery = wrap(recovery)
 
     def stats(self, *args, **kwargs):
-        return self._stats
+        return self._stats()
 
     def info(self, *args, **kwargs):
-        return self._info
+        return self._info()
 
     def recovery(self, *args, **kwargs):
-        return self._recovery
+        return self._recovery()
 
+def wrap(it):
+    return it if callable(it) else ResponseSupplier(it)
+
+class ResponseSupplier:
```

Comment:
PEP8: 2 blank lines here

Reply:
Addressed in https://github.com/elastic/rally/pull/730/commits/3cfa5450f1172fa8744d2e98ad22a119ce991ebf.